gems/aws-sdk-core/lib/aws-sdk-core/arn.rb

```diff
@@ -29,18 +29,22 @@ module AWS::SDK::Core
   #
   # @see https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#genref-arns
   class ARN
+    # Raised when ARN string input doesn't follow the standard:
+    # https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#genref-arns
+    class InvalidARNError < RuntimeError; end
```

Comment:
I think maybe this belongs in ARN parser?

Reply:
Moved.